x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/templates_form/templates_form.tsx

```diff
@@ -0,0 +1,126 @@
+  defaultValue?: MappingsTemplates;
+}
+
+const stringifyJson = (json: { [key: string]: any }) =>
+  Array.isArray(json) ? JSON.stringify(json, null, 2) : '[\n\n]';
+
+const formSerializer: SerializerFunc<MappingsTemplates> = formData => {
+  const { dynamicTemplates } = formData;
+
+  let parsedTemplates;
+  try {
+    parsedTemplates = JSON.parse(dynamicTemplates);
+
+    if (!Array.isArray(parsedTemplates)) {
+      // User provided an object, but we need an array of objects
+      parsedTemplates = [parsedTemplates];
+    }
+  } catch {
+    parsedTemplates = [];
+  }
+
+  return {
+    dynamic_templates: parsedTemplates,
+  };
+};
+
+const formDeserializer = (formData: { [key: string]: any }) => {
+  const { dynamic_templates } = formData;
+
+  return {
+    dynamicTemplates: stringifyJson(dynamic_templates),
+  };
+};
+
+export const TemplatesForm = React.memo(({ defaultValue }: Props) => {
+  const didMountRef = useRef(false);
+
+  const { form } = useForm<MappingsTemplates>({
+    schema: templatesFormSchema,
+    serializer: formSerializer,
+    deserializer: formDeserializer,
+    defaultValue,
+  });
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const subscription = form.subscribe(updatedTemplates => {
+      dispatch({ type: 'templates.update', value: { ...updatedTemplates, form } });
+    });
+    return subscription.unsubscribe;
+  }, [form]);
+
+  useEffect(() => {
+    if (didMountRef.current) {
+      // If the defaultValue has changed (it probably means that we have loaded a new JSON)
+      // we need to reset the form to update the fields values.
+      form.reset({ resetValues: true });
+    } else {
+      // Avoid reseting the form on component mount.
+      didMountRef.current = true;
+    }
+  }, [defaultValue]);
+
+  useEffect(() => {
+    return () => {
+      // On unmount => save in the state a snapshot of the current form data.
+      dispatch({ type: 'templates.save' });
+    };
+  }, []);
+
+  return (
+    <>
+      <EuiText size="s" color="subdued">
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.dynamicTemplatesDescription"
+          defaultMessage="Use dynamic templates to define custom mappings that can be applied to dynamically added fields. {docsLink}"
+          values={{
+            docsLink: (
+              <EuiLink href={documentationService.getDynamicTemplatesLink()} target="_blank">
+                {i18n.translate('xpack.idxMgmt.mappingsEditor.dynamicTemplatesDocumentationLink', {
+                  defaultMessage: 'Learn more.',
+                })}
+              </EuiLink>
+            ),
+          }}
+        />
+      </EuiText>
+      <EuiSpacer size="m" />
+      <Form form={form} isInvalid={form.isSubmitted && !form.isValid} error={form.getErrors()}>
+        <UseField
+          path="dynamicTemplates"
+          component={JsonEditorField}
+          componentProps={{
+            euiCodeEditorProps: {
+              height: '400px',
```

Comment:
This is a pretty narrow window, and this object can get fairly long. How would you feel about bumping this up just a bit, e.g. to 600px?